src/main/java/edu/kis/powp/jobs2d/command/visitor/CommandCanvasVisitor.java

```diff
@@ -0,0 +1,44 @@
+package edu.kis.powp.jobs2d.command.visitor;
+
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.ICompoundCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.jobs2d.drivers.transformation.Point;
+
+import java.util.Iterator;
+
+public class CommandCanvasVisitor implements Visitor {
+
+    private final Canvas canvas;
+    private boolean allPointsOnCanvas = true;
+
+    public CommandCanvasVisitor(Canvas canvas) {
+        this.canvas = canvas;
+    }
+
+    @Override
+    public void visitOperateToCommand(OperateToCommand operateToCommand) {
+        Point pointToCheck = operateToCommand.getPoint();
+        allPointsOnCanvas = !canvas.checkIfPointIsOnCanvas(pointToCheck);
+    }
+
+    @Override
+    public void visitSetPositionCommand(SetPositionCommand setPositionCommand) {
+        Point pointToCheck = setPositionCommand.getPoint();
+        allPointsOnCanvas = !canvas.checkIfPointIsOnCanvas(pointToCheck);
+    }
+
+    @Override
+    public void visitICompoundCommand(ICompoundCommand iCompoundCommand) {
+        Iterator<DriverCommand> iterator = iCompoundCommand.iterator();
+        while (iterator.hasNext()) {
+            DriverCommand driverCommand = iterator.next();
+            driverCommand.accept(this);
+        }
+    }
+
+    public boolean checkIfAllPointsOnCanvas(){
```

Comment:
maybe add Status suffix at the and of method (because the method does not check, but returns the status of the previous call).